### DISCUSSION TITLE
Allow generating Heapusage reports on demand

### DISCUSSION THREAD

**Discussion by rjpdasilva** [d99kris/heapusage#13] [Ideas]
I think the idea is better understood by some actual changes proposal, so please refer to #12.

This is just a proposal, as I might not be aware of all the implications on platforms other than Linux, on which they were actually used, tested and proven quite useful.

Looking forward from hearing some feedback.

**Comment by d99kris** [comment]
Hi @rjpdasilva - thanks for proposing some improvements and providing implementation as well. 👍 
I will look into this and get back.

**Comment by rjpdasilva** [comment]
Implemented.
